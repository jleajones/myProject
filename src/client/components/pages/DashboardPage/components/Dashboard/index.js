import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Container from '@material-ui/core/Container';
import makeStyles from '@material-ui/core/styles/makeStyles';

import EnhancedPaper from '@shared/EnhancedPaper';

const useStyles = makeStyles({
  pageview: {
    borderColor: 'rgba(63, 81, 188, 0.35)'
  },
  interaction: {
    borderColor: 'rgba(63, 81, 188, 0.55)'
  }
});

const Dashboard = ({ data }) => {
  const classes = useStyles();
  return (
    <Container>
      <Grid container spacing={2}>
        {data.map(item => {
          const keys = Object.keys(item);
          const itemId = item.id || item.eventId;

          return (
            <Grid item key={itemId} xs={12}>
              <EnhancedPaper
                classFromParent={{
                  child: classes[item.eventType]
                }}
                label={`${itemId}`}
                category="hero"
              >
                {/* eslint-disable-next-line array-callback-return,consistent-return */}
                {keys.map(key => {
                  const value = item[key];
                  const isObject = typeof value === 'object';
                  if (
                    (value && !isObject) ||
                    (value && isObject && Object.keys(value).length > 0)
                  ) {
                    return (
                      <Typography key={`${itemId}-${key}`}>
                        <b>{key}</b>{' '}
                        {isObject ? JSON.stringify(item[key]) : item[key]}
                      </Typography>
                    );
                  }
                })}
              </EnhancedPaper>
            </Grid>
          );
        })}
      </Grid>
    </Container>
  );
};

Dashboard.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Dashboard;
