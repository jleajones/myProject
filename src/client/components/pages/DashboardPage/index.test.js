import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import Dashboard from '.';

describe('<Dashboard />', () => {
  const fetchEvents = jest.fn();
  const store = configureStore();
  const initialState = {
    analytics: {
      events: [
        {
          id: 1,
          username: 'HarveyD',
          email: 'harvey@email.com',
          joined: '2019-07-07T04:45:04.524Z',
          last_logged_in: '2019-07-07T04:45:04.524Z'
        },
        {
          id: 3,
          username: 'jleajones',
          email: 'jleajones@email.com',
          joined: '2019-08-12T19:24:21.942Z',
          last_logged_in: '2019-08-12T19:24:21.942Z'
        },
        {
          id: 4,
          username: 'darmstr2',
          email: 'darmstr2@email.com',
          joined: '2019-08-12T19:24:39.744Z',
          last_logged_in: '2019-08-12T19:24:39.744Z'
        }
      ]
    },
    users: {
      data: [
        {
          id: 1,
          username: 'HarveyD',
          email: 'harvey@email.com',
          joined: '2019-07-07T04:45:04.524Z',
          last_logged_in: '2019-07-07T04:45:04.524Z'
        },
        {
          id: 3,
          username: 'jleajones',
          email: 'jleajones@email.com',
          joined: '2019-08-12T19:24:21.942Z',
          last_logged_in: '2019-08-12T19:24:21.942Z'
        },
        {
          id: 4,
          username: 'darmstr2',
          email: 'darmstr2@email.com',
          joined: '2019-08-12T19:24:39.744Z',
          last_logged_in: '2019-08-12T19:24:39.744Z'
        }
      ]
    }
  };
  test('Renders component without crashing', () => {
    const component = shallow(
      <Dashboard fetchEvents={fetchEvents} store={store(initialState)} />
    );

    expect(component).toMatchSnapshot();
  });
});
