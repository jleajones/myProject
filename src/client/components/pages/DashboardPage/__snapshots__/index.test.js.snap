// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> Renders component without crashing 1`] = `
<ContextProvider
  value={null}
>
  <DashboardPage
    events={
      Array [
        Object {
          "email": "harvey@email.com",
          "id": 1,
          "joined": "2019-07-07T04:45:04.524Z",
          "last_logged_in": "2019-07-07T04:45:04.524Z",
          "username": "HarveyD",
        },
        Object {
          "email": "jleajones@email.com",
          "id": 3,
          "joined": "2019-08-12T19:24:21.942Z",
          "last_logged_in": "2019-08-12T19:24:21.942Z",
          "username": "jleajones",
        },
        Object {
          "email": "darmstr2@email.com",
          "id": 4,
          "joined": "2019-08-12T19:24:39.744Z",
          "last_logged_in": "2019-08-12T19:24:39.744Z",
          "username": "darmstr2",
        },
      ]
    }
    fetchEvents={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
